build_system/src/clean.rs

```diff
@@ -0,0 +1,56 @@
+use crate::utils::remove_file;
+
+use std::fs::remove_dir_all;
+
+fn check_args() -> Result<Option<()>, String> {
+    for arg in std::env::args().skip(2) {
+        match arg.as_str() {
+            "--help" => {
+                usage();
+                return Ok(None);
+            }
+            a => return Err(format!("Unknown argument `{a}`")),
```

Comment:
I prefer the old syntax, but I guess if this uses clippy, this might throw a warning:
```suggestion
            arg => return Err(format!("Unknown argument `{}`", arg)),
```
Hopefully, GitHub will highlight those inline arguments in the future.